### PR TITLE
Adding interfaces

### PR DESCRIPTION
**Pull request by rosemwahlin**
This is an initial implementation – I based it on our test files instead of a thorough reading of the spec like I did for the other IDL. Let me know if this will be presentable for Thursday.

### CODE REVIEW

File: syntaxes/omg.idl.tmLanguage.json [hunk 1]
```diff
@@ -142,7 +142,8 @@
         "type_dcl": {
             "patterns": [
                 {"include": "#constr_type_dcl"},
-                {"include": "#typedef_dcl"}
+                {"include": "#typedef_dcl"},
+                {"include": "#interface_def"}
             ]
```
Comment (fgaranda) [inline syntaxes/omg.idl.tmLanguage.json:149]: I think we need to add also the ability to declare exceptions.

File: syntaxes/omg.idl.tmLanguage.json [hunk 2]
```diff
@@ -572,11 +573,109 @@
...
+        "op_type_spec": {
+            "patterns": [
+                {"include": "#void_type"},
+                {"include": "#const_type"},
+                {"include": "#template_type_spec"}
+            ]
+        },
+        "parameter_dcl": {
+            "name": "meta.storage.operation.parameter.omg.idl",
+            "begin": "(in|out|inout)\\b",
```
Comment (fgaranda) [inline syntaxes/omg.idl.tmLanguage.json]: Is this mandatory? If you don't specify anything I think it defaults to `in`

File: syntaxes/omg.idl.tmLanguage.json [hunk 3]
```diff
@@ -142,7 +142,8 @@
         "type_dcl": {
             "patterns": [
                 {"include": "#constr_type_dcl"},
-                {"include": "#typedef_dcl"}
+                {"include": "#typedef_dcl"},
+                {"include": "#interface_def"}
```
Comment (fgaranda) [inline syntaxes/omg.idl.tmLanguage.json]: We should add a declaration to allow also forward declaration of interfaces.

File: syntaxes/omg.idl.tmLanguage.json [hunk 4]
```diff
@@ -572,11 +573,109 @@
             "name": "annotation.omg.idl",
             "patterns": [{
                 "name": "storage.type.annotation.omg.idl",
-                "match": "@\\w+(\\(.*\\)){0,1}"
+                "match": "@\\w+\\s*(\\(.*\\)){0,1}"
             }]
+        },
+        "interface_def": {
+            "name": "meta.storage.interface.omg.idl",
+            "begin": "(interface)\\s+(\\w+)\\s*\\{",
```
Comment (fgaranda) [inline syntaxes/omg.idl.tmLanguage.json]: Interfaces can inherit from other interfaces; I think you have that syntax in structs.

**Review by fgaranda — CHANGES_REQUESTED** [review]

File: syntaxes/omg.idl.tmLanguage.json (hunk 1 above)
Comment (rosemwahlin) [reply syntaxes/omg.idl.tmLanguage.json:149]: Done.

File: syntaxes/omg.idl.tmLanguage.json (hunk 2 above)
Comment (rosemwahlin) [reply syntaxes/omg.idl.tmLanguage.json]: The spec doesn't say anything about this being optional, but I've made it optional nonetheless, so we can support a default value.

File: syntaxes/omg.idl.tmLanguage.json (hunk 3 above)
Comment (rosemwahlin) [reply syntaxes/omg.idl.tmLanguage.json]: Done.

File: syntaxes/omg.idl.tmLanguage.json (hunk 4 above)
Comment (rosemwahlin) [reply syntaxes/omg.idl.tmLanguage.json]: Done.

**Review by fgaranda — APPROVED** [review]

**Pull request merged**